### DISCUSSION TITLE
GUI

### DISCUSSION THREAD

**Discussion by oskardotglobal** [winapps-org/winapps#6] [Ideas]
Not a priority so I'll leave this here for now. We should definetly consider a GUI where some things can be done easily, such as:
* accessing the vm fully
* creating and managing the vma
* installing and managing apps
* adding the apps as `.desktop` files
* listing, grouping and running apps

**Comment by LDprg** [comment]
We also need to choose a rust gui libary (for the rewrite codebase)
There are some options available:
- Tauri (like electron)
- Bindings (QT, GTK)
- pure rust libaries


A good collection of libaries:
https://areweguiyet.com/

I would prefer a pure rust libary or the gtk bindings, but I haven't worked with them before. (Only worked with Tauri but I don't think it fits our purpose)

**Reply by ludgie** [reply]
Hello,

I think I'd be better to write a toolkit agnostic backend, for then anyone (or the org/project itself) write a GUI with GTK or Qt or Tauri...

**Reply by oskardotglobal** [reply]
The idea was to have the `winapps` crate contain all core functionality and then have `winapps-cli` build upon that. This was also planned for the GUI.

**Reply by LDprg** [reply]
Yeah exactly. The winapps crate itself is just a library allowing cli, gui and other kind of stuff to be based on. However I think an official implementation of the gui is probably  the best.
